blocks/carousel/carousel.js

```diff
@@ -0,0 +1,281 @@
+      x = newValue;
+    }
+  };
+
+  const updateButtonVisibility = () => {
+    if (x > -40) {
+      leftButton.classList.add('hidden');
+    } else {
+      leftButton.classList.remove('hidden');
+    }
+    if (x < -maxScroll + 40) {
+      rightButton.classList.add('hidden');
+    } else {
+      rightButton.classList.remove('hidden');
+    }
+  };
+  updateButtonVisibility();
+
+  // drag logic starting with initialization for mobile
+  let previousTouch;
+
+  const mouseMoveHandler = (e) => {
+    if (!pressed) return;
+    if (!e?.touches?.[0]) {
+      e.preventDefault();
+    }
+    addToX(e.movementX);
+    carouselFrame.style.transform = `translateX(${x}px)`;
+    // prettier-ignore
+  };
+
+  const touchMoveHandler = (e) => {
+    const touch = e.touches[0];
+    if (previousTouch) {
+      e.movementX = touch.pageX - previousTouch.pageX;
+      mouseMoveHandler(e);
+    }
+    previousTouch = touch;
+  };
+
+  // mouse drag handlers
+  const mouseUpHandler = () => {
+    pressed = false;
+    carouselWrapper.classList.remove('grabbed');
+    roundX();
+    carouselFrame.style.transform = `translateX(${x}px)`;
+    updateButtonVisibility();
+    // cleanup
+    window.removeEventListener('mouseup', mouseUpHandler);
+    window.removeEventListener('touchend', mouseUpHandler);
+
+    window.removeEventListener('mousemove', mouseMoveHandler);
+    window.removeEventListener('touchmove', touchMoveHandler);
+  };
+
+  const mouseDownHandler = (e) => {
+    if (e?.touches?.[0]) {
+      [previousTouch] = e.touches;
+    }
+    carouselWrapper.classList.add('grabbed');
+    pressed = true;
+    refreshMaxScroll();
+
+    window.addEventListener('mouseup', mouseUpHandler);
+    window.addEventListener('touchend', mouseUpHandler);
+
+    window.addEventListener('mousemove', mouseMoveHandler);
+    window.addEventListener('touchmove', touchMoveHandler);
+  };
+  carouselWrapper.onmousedown = mouseDownHandler;
+  carouselWrapper.addEventListener('touchstart', mouseDownHandler);
+
+  // button logic
+  const rightOnClick = () => {
+    addToX(-carouselFrame.getBoundingClientRect().width);
+    roundX();
+    carouselFrame.style.transform = `translateX(${x}px)`;
+    updateButtonVisibility();
+  };
+
+  const leftOnClick = () => {
+    addToX(carouselFrame.getBoundingClientRect().width);
+    roundX();
+    carouselFrame.style.transform = `translateX(${x}px)`;
+    updateButtonVisibility();
+  };
+
+  rightButton.onclick = rightOnClick;
+  leftButton.onclick = leftOnClick;
+
+  // preventing drag from triggering a page switch & focus logic
+  const cardFocusHandler = (cardLink, index) => {
+    refreshMaxScroll();
+    const width = cardLink.getBoundingClientRect().width + gapOffset;
+    x = width * -index;
+    roundX(width);
+    updateButtonVisibility();
+    carouselFrame.style.transform = `translateX(${x}px)`;
+  };
+
+  [...carouselCardLinks].forEach((anchor, index) => {
+    const clickLocation = {};
+    const keyUpHandler = (e) => {
```

Comment:
It would be nice if you could make the card link keyboard accessible again eg  when you press enter when the link is focused, it navigates to the url.

Reply:
links can now be opened with Enter while focused